crates/re_log_types/src/path/data_path.rs

```diff
@@ -0,0 +1,41 @@
+use re_types::{components::InstanceKey, ComponentName};
+
+use crate::EntityPath;
+
+/// A general path to some data.
+///
+/// This always starts with an [`EntityPath`], followed
+/// by an optional [`InstanceKey`], followed by an optional [`ComponentName`].
+///
+/// For instance:
+///
+/// * `points`
+/// * `points.Color`
+/// * `points[#42]`
+/// * `points[#42].Color`
+pub struct DataPath {
```

Comment:
just noticed that `InstancePath` is the odd one out - it's in re_data_store unlike the others and has an overlapping relationship with `DataPath`. Might be nice to put them all next to each other in a follow-up PR and have an overview over all types of paths.

Reply:
It is odd because it depends on `InstanceKey` which is a code-genned thing that lives in `re_types`. I guess we could move all paths to `re_types` though at some point soon.